controller/request.go

```diff
@@ -0,0 +1,200 @@
+package controller
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+
+	"github.com/giantswarm/inago/fleet"
+)
+
+func DefaultRequestConfig() RequestConfig {
+	newConfig := RequestConfig{
+		Group:    "",
+		SliceIDs: []string{},
+	}
+
+	return newConfig
+}
+
+type RequestConfig struct {
+	// Group represents the plain group name without any slice expression.
+	Group string
+
+	// SliceIDs contains the IDs to create. IDs can be "1", "first", "whatever",
+	// "5", etc..
+	SliceIDs []string
+}
+
+// Request represents a controller request. This is used to process some action
+// on the controller.
+type Request struct {
+	RequestConfig
+
+	// Units represents a list of unit files that is supposed to be extended
+	// using the provided slice IDs.
+	Units []Unit
+}
+
+func NewRequest(config RequestConfig) Request {
+	req := Request{
+		RequestConfig: config,
+		Units:         []Unit{},
+	}
+
+	return req
+}
+
+var unitExp = regexp.MustCompile("@.service")
+
+// ExtendSlices extends unit files with respect to the given slice IDs. Having
+// slice IDs "1" and "2" and having unit files "foo@.service" and
+// "bar@.service" results in the following extended unit files.
+//
+// 	 foo@1.service
+// 	 bar@1.service
+// 	 foo@2.service
+// 	 bar@2.service
+//
+func (r Request) ExtendSlices() (Request, error) {
+	if len(r.SliceIDs) == 0 {
+		return r, nil
+	}
+
+	var newUnits []Unit
+	for _, sliceID := range r.SliceIDs {
+		for _, unit := range r.Units {
+			newUnit := unit
+			// TODO fix extension
+			newUnit.Name = unitExp.ReplaceAllString(newUnit.Name, fmt.Sprintf("@%s.service", sliceID))
+			newUnits = append(newUnits, newUnit)
+		}
+	}
+	r.Units = newUnits
+
+	return r, nil
+}
+
+func (c controller) readUnitFiles(dir string) (map[string]string, error) {
```

Comment:
My thoughts went into the direction of what the request need to look like. The controller requires a request. The request needs to hold certain information. Thinking about a server client architecture this makes not that much sense, but we are not there. Thus I opted for consistency for the request creation, even this is not future prove.